blocks/editable/index.js

```diff
@@ -177,10 +178,34 @@ export default class Editable extends wp.element.Component {
 		return true;
 	}
 
+	isEndOfEditor() {
+		const range = this.editor.selection.getRng();
+		if ( range.endOffset !== range.endContainer.textContent.length || ! range.collapsed ) {
+			return false;
+		}
+		const start = range.endContainer;
+		const body = this.editor.getBody();
+		let element = start;
+		while ( element !== body ) {
+			const child = element;
+			element = element.parentNode;
+			if ( element.lastChild !== child ) {
+				return false;
+			}
+		}
+		return true;
+	}
+
 	onKeyDown( event ) {
-		if ( this.props.onMerge && event.keyCode === KEYCODE_BACKSPACE && this.isStartOfEditor() ) {
+		if (
+			this.props.onMerge && (
+				( event.keyCode === KEYCODE_BACKSPACE && this.isStartOfEditor() ) ||
+				( event.keyCode === KEYCODE_DELETE && this.isEndOfEditor() )
+			)
+		) {
+			const forward = event.keyCode === KEYCODE_DELETE;
 			this.onChange();
-			this.props.onMerge( this.editor.getContent() );
+			this.props.onMerge( forward );
```

Comment:
>Imagine a block having multiple editables.

Ah, that's a good point. I hadn't considered this.

>Could we provide more "metadata" on how the block uses the editable

Interesting idea. I think at the point where we're saying a block implementer needs to be concerned with merge behavior (and it's fairly obvious they do), I'm not as much concerned about accepting and applying the `onMerge` prop handling anymore.

It could be slightly more optimized if Editable were able to handle this by hints, but I'm not sure a clean way we can connect the Editable to the editor's store without tying it directly to the editor (not so great).